src/jose.c

```diff
@@ -922,21 +923,51 @@ static apr_byte_t oidc_jose_zlib_uncompress(apr_pool_t *pool, const char *input,
 	zlib.opaque = Z_NULL;
 	zlib.avail_in = (uInt) input_len;
 	zlib.next_in = (Bytef*) input;
-	*output = apr_pcalloc(pool, input_len * 4);
-	zlib.avail_out = (uInt) (input_len * 4);
-	zlib.next_out = (Bytef*) (*output);
+	zlib.total_out = 0;
+	size_t uncompLength = OIDC_CJOSE_UNCOMPRESS_CHUNK;
+	char *uncomp = (char *) calloc(sizeof (char), uncompLength + 1);
```

Comment:
why not use apr_pcalloc hare and leave memory management up to apr?

Reply:
I thought that apr might allocate a lot of memory because it does not free().
If the target is a cookie value and there is no need to uncompress a large size, I would use apr.

Reply:
apr_pcalloc(r->pool,<>) would allocate the same size of memory, and the request pool is freed after the request completes, together with everything else allocated whilst processing the request

Reply:
I see, change to apr.